src/osd/OSD.cc

```diff
@@ -860,7 +860,7 @@ bool OSDService::need_fullness_update()
   return want != cur;
 }
 
-bool OSDService::_check_full(s_names type, ostream &ss) const
```

Comment:
how about passing the return value of `get_parent()->get_dpp()` to this method? so we can preserve the `const` specifier.

Reply:
Hi Kefu,

I think that would work except for the calls to check_backfill_full() in PG.cc since everything else has access to a PGBackend::Listener.  FWIW, I left the original douts partially in place to still get the pg prefix info, but I agree it's not ideal.  Keeping these methods const and using get_dpp() would be preferable.

Reply:
oh! PG implements DoutPrefixProvider.  Haha. :)